lightning/src/offers/static_invoice.rs

```diff
@@ -72,6 +74,88 @@ struct InvoiceContents {
 	message_paths: Vec<BlindedPath>,
 }
 
+/// Builds a [`StaticInvoice`] from an [`Offer`].
+///
+/// See [module-level documentation] for usage.
+///
+/// [`Offer`]: crate::offers::offer::Offer
+/// [`InvoiceRequest`]: crate::offers::invoice_request::InvoiceRequest
+/// [`Bolt12Invoice`]: crate::offers::invoice::Bolt12Invoice
+/// This is not exported to bindings users as builder patterns don't map outside of move semantics.
+pub struct StaticInvoiceBuilder<'a> {
+	offer_bytes: &'a Vec<u8>,
+	invoice: InvoiceContents,
+	keys: KeyPair,
```

Comment:
Could we derive the keys from the offer metadata rather than requiring that the caller pass them?